scripts/make_help.py

```diff
@@ -1,57 +1,94 @@
 from pathlib import Path
 from argparse import ArgumentParser
-
-
-YELLOW = "\033[33m"
-BLUE = "\033[34m"
-RESET = "\033[0m"
-
-
-def parse_makefile(file: Path, res: dict[str, list[tuple[str, str]]]):
-    with open(file, "r") as f:
-        contents = f.readlines()
-        contents = map(lambda x: x.strip(), contents)
-        contents = filter(lambda x: x.startswith("## "), contents)
-        contents = map(lambda x: x.lstrip("## "), contents)
-
-        for line in contents:
-            data = line.split(":")
-            assert len(data) == 3
-
-            grp = data[0].strip()
-            cmd = data[1].strip()
-            val = data[2].strip()
-
-            if grp in res:
-                res[grp].append((cmd, val))
-            else:
-                res[grp] = [(cmd, val)]
-
-
-def create_help(files: list[Path]) -> dict[str, list[tuple[str, str]]]:
-    res = dict()
+from typing import Dict, List, Tuple
+from collections import defaultdict
+
+# ANSI color codes
+class Colors:
+    YELLOW = "\033[33m"
+    BLUE = "\033[34m"
+    RESET = "\033[0m"
+
+def parse_makefile(file: Path) -> Dict[str, List[Tuple[str, str]]]:
+    """Parse a Makefile and extract help information.
+    
+    Args:
+        file: Path to the Makefile
+    Returns:
+        Dictionary containing grouped help information
+    """
+    result = defaultdict(list)
+    try:
```

Comment:
No point in doing this. This is not a library or even a particularly useful file. Just a way for pretty help message.

It is expected to be called from the `Makefile`. As such, the `Makefile` in question will always exist.